src/memkind_arena.c

```diff
@@ -544,7 +544,10 @@ MEMKIND_EXPORT void *memkind_arena_calloc(struct memkind *kind, size_t num,
     err = kind->ops->get_arena(kind, &arena, size);
     if (MEMKIND_LIKELY(!err)) {
         result = jemk_mallocx_check(num * size,
-                                    MALLOCX_ARENA(arena) | MALLOCX_ZERO | get_tcache_flag(kind->partition, size));
+                                    MALLOCX_ARENA(arena) | get_tcache_flag(kind->partition, size));
+        if (MEMKIND_LIKELY(result)) {
+            memset(result, 0, size);
```

Comment:
hmm.. I think we should only do this for kinds that use file-based mappings.

Reply:
Done. 
I leave old implementation responsible for arena calloc function untouched.
I introduce separate function, which handle file-backed memory case.